### PR TITLE
Feature/sensor cfg

### PR DESCRIPTION
**Pull request by brianpax**
Abstracted sensor configuration function in sensor driver library

### CODE REVIEW

**Comment by vrahane** [comment]
Hi @brianpax, code looks fine to me, I think you should rebase against apache/master instead of runtimeinc/pax-master-fork and push the changes. The objective is to remove the Merge pull request commits from the commit history and also get the code up-to-date.

File: hw/drivers/sensors/ms5837/src/ms5837.c [hunk 1]
```diff
@@ -422,7 +432,7 @@ ms5837_read_eeprom(struct sensor_itf *itf, uint16_t *coeff)
     rc = ms5837_crc_check(payload, (payload[MS5837_IDX_CRC] & 0xF000) >> 12);
     if (rc) {
         rc = SYS_EINVAL;
-        MS5837_ERR("Failure in CRC, 0x%02X\n", payload[idx]);
+        // MS5837_ERR("Failure in CRC, 0x%02X\n", payload[idx]);
```
Comment (rymanluk) [inline hw/drivers/sensors/ms5837/src/ms5837.c]: Looks like something what slipped in.

File: hw/sensor/include/sensor/sensor.h [hunk 2]
```diff
@@ -326,6 +336,7 @@ typedef int (*sensor_set_trigger_thresh_t)(struct sensor *, sensor_type_t,
 struct sensor_driver {
     sensor_read_func_t sd_read;
     sensor_get_config_func_t sd_get_config;
+    sensor_config_func_t sd_config;
```
Comment (rymanluk) [inline hw/sensor/include/sensor/sensor.h]: Since we have sensor_get_config_func_t I suggest to name the new one sensor_set_config_func_t

**Comment by brianpax** [comment]
Made requested changes, rebased onto apache/master, commit message fixed

**Comment by vrahane** [comment]
@samandersonpax I think changes to libc/baselibc/src/tinyprintf.c should be a separate PR. Rest looks good.

**Comment by brianpax** [comment]
Sorry @vrahane, looks like that commit snuck in when I rebased. This covers only the sensor changes.

**Comment by vrahane** [comment]
Not a problem. LGTM now. After @rymanluk finishes his review, I will merge it.

**Pull request merged**